src/runtime/simulator.cc

```diff
@@ -482,12 +486,32 @@ ParallelConfig Op::view_to_pc(MachineView const &view) const {
   return config;
 }
 
+/**
+ * @brief Try to get the operator's parameters.
+ *
+ * @note This function can be improved to avoid forcibly casting the const Op*
+ * to a specific operator pointer type. Ideally, this should be implemented as a
+ * dynamic dispatch on the Op base class (i.e. define a get_params() virtual
+ * function on Op class).
+ *
+ * To avoid the forcibly casting from a const Op pointer to a specific operator
```

Comment:
I was meaning it might not be perfect to cast the pointer type like this. We should use dynamic cast to move along the inheritance tree. But since the input is const, we need to cast out the const qualifier first. Still, I think a better way for this is to have a base class function to get params and rely on dynamic dispatch. But no need to make the change now as the current design also works.